### PR TITLE
Fixes #17079: Test files are not correctly clean

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17079

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/domain/reports/StatusReportTest.scala [hunk 1]
```diff
@@ -297,13 +295,10 @@ class StatusReportTest extends Specification {
 
       "run fast enough to sum" in {
         ComplianceLevel.sum(initData.map( x => ComplianceLevel.compute(x._2)))
-
-        val source = initData.map(x => (x._1, ComplianceLevel.compute(x._2))).map( _._2)
         val t0 = System.nanoTime
 
         for (i <- 1 to 100) {
           val t0_0 = System.nanoTime
-          val result = ComplianceLevel.sum(source)
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/domain/reports/StatusReportTest.scala:306]: these should not be removed. They are used to warm the JVM, plus are not at all related to this ticket

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/domain/reports/StatusReportTest.scala:306]: Those variable are unused, and really look like dead code

Comment (fanf) [reply webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/domain/reports/StatusReportTest.scala:306]: @ncharles if they are useful, we need a way to mark them as that. I regularly remove things that lead to warning or looks like dead code. At minima, they need to have: 
- a comment to explain why they are here,
- a silent annotation to remove warning. 
Rudder should build with 0 warning

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/hooks/HooksTest.scala [hunk 2]
```diff
@@ -74,7 +75,9 @@ class HooksTest() extends Specification with AfterAll {
   }
 
   override def afterAll(): Unit = {
-   // tmp.delete()
+    if(System.getProperty("tests.clean.tmp") != "false") {
+      File("/tmp/rudder-test-hook/").delete()
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/hooks/HooksTest.scala:79]: In think @amousset will be very happy (test will now be cleaned up ....)

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**